Tools/ardupilotwaf/git_submodule.py

```diff
@@ -111,8 +112,62 @@ def u(s):
     def __str__(self):
         return 'Submodule update: %s' % self.submodule
 
+def check_folder(path_list=[]):
+    for d in Utils.to_list(path_list):
+        listdir = os.listdir(d)
+        for f in listdir(d):
+            if f.startsith('.'):
+                listdir.remove()
+        if len(f) == 0:
+            raise
+    return f
+
+def module_is_uninitialised(d):
+    listdir = os.listdir(d)
+    listdir_trimmed = [i for i in listdir if not i.startswith('.')]
+    print("listdir_trimmed", listdir_trimmed)            
+    if len(listdir_trimmed) == 0:
+        return True
+
+    return False
+
+
 def configure(cfg):
+    print("============CONFIGURE===================")
     cfg.find_program('git')
+    path = cfg.srcnode.abspath()+'/modules/'
+    check_modules = ['ChibiOS', 'gbenchmark', 'gtest', 'libcanard', 'mavlink', 'uavcan']
+    missing_modules = []
+
+    for i in check_modules:
+        path_module = (path+i)
+        print("CHECK MODULE", i)
+        if module_is_uninitialised(path_module) == True:
+            missing_modules.append(i)
+
+    print("MISSING_MODULES",missing_modules)
+
+    for i in missing_modules:
+        try:
+            subprocess.check_call(['git', 'submodule', 'update', '--init', (path+i)])
+        except:
+            cfg.fatal('Missing submodules!')
```

Comment:
Does this squash all of the useful output from git?

Any reason to catch the exception?

Reply:
Not sure if it is actually useful tbh. I think it is mainly to make sure that either the module gets updated or an error is thrown that is somewhat indicative as to what might be the problem. 

Would it be better to just leave out the try-except all together? (Since one can assume git will also output something useful?)